util.c

```diff
@@ -2494,12 +2494,7 @@ Perl_my_setenv(pTHX_ const char *nam, const char *val)
         char *new_env = S_env_alloc(NULL, nlen, vlen, 2, 1);
         /* all that work just for this */
         my_setenv_format(new_env, nam, nlen, val, vlen);
-#    ifndef WIN32
         putenv(new_env);
-#    else
-        PerlEnv_putenv(new_env);
-        safesysfree(new_env);
-#    endif
```

Comment:
This has three problems.

- On Win32 this will leak memory, POSIX-like* putenv holds onto the pointer and includes it in the environment, but this isn't true for Win32.
- this bypasses the thread safety of PerlEnv_putenv() added in 24f3e849b5c on POSIX-likes
- the putenv() isn't replaced on Win32, so it calls the CRT putenv() rather than PerlEnv_putenv() aka win32_putenv(), which probably explains the Win32 test failures.  Replacing the putenv() with PerlEnv_putenv() made one of the failing tests pass.


* putenv() hasn't been POSIX for a while